tests/unit/serve/runtimes/worker/test_worker_runtime.py

```diff
@@ -426,3 +429,54 @@ def start_runtime(args, cancel_event):
     runtime_thread.join()
 
     assert not AsyncNewLoopRuntime.is_ready(f'{args.host}:{args.port}')
+
+
+@pytest.mark.slow
+@pytest.mark.timeout(10)
+def test_error_in_worker_runtime_with_exit_on_exceptions(monkeypatch):
+    args = set_pod_parser().parse_args(['--exit-on-exceptions', 'RuntimeError'])
+
+    cancel_event = multiprocessing.Event()
+
+    def fail(*args, **kwargs):
+        raise RuntimeError('intentional error')
+
+    monkeypatch.setattr(DataRequestHandler, 'handle', fail)
+
+    def start_runtime(args, cancel_event):
+        with WorkerRuntime(args, cancel_event=cancel_event) as runtime:
+            runtime.run_forever()
+
+    runtime_thread = Process(
+        target=start_runtime,
+        args=(args, cancel_event),
+        daemon=True,
+    )
+    runtime_thread.start()
+
+    assert AsyncNewLoopRuntime.wait_for_ready_or_shutdown(
+        timeout=5.0,
+        ctrl_address=f'{args.host}:{args.port}',
+        ready_or_shutdown_event=Event(),
+    )
+
+    target = f'{args.host}:{args.port}'
+    try:
+        with grpc.insecure_channel(
+            target,
+            options=GrpcConnectionPool.get_default_grpc_options(),
+        ) as channel:
+            stub = jina_pb2_grpc.JinaSingleDataRequestRPCStub(channel)
+            response, call = stub.process_single_data.with_call(
+                _create_test_data_message()
+            )
+
+        assert response.header.status.code == jina_pb2.StatusProto.ERROR
+        assert 'is-error' in dict(call.trailing_metadata())
+    except:
+        pass
```

Comment:
1. can't we use the grpc client used by the gateway runtime ?
2. I believe asserts shouldn't be wrapped in a try-catch

Reply:
1. I was using the first example as my sample to create the new test. I checked the other tests and replaced the call with the available helper method.
2. I removed the try-catch since the Runtime gracefully handles the exception and returns a generic error request. I wasn't able to validate before because of I couldn't run the test locally. Anyway now I can run the test partially and verify which works the best.